internal/backends/android-activity/lib.rs

```diff
@@ -0,0 +1,663 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+#![doc = include_str!("README.md")]
+#![doc(html_logo_url = "https://slint.dev/logo/slint-logo-square-light.svg")]
+#![cfg(target_os = "android")]
+
+pub use android_activity;
+use android_activity::input::{InputEvent, KeyAction, Keycode, MotionAction, MotionEvent};
+pub use android_activity::AndroidApp;
+use android_activity::{InputStatus, MainEvent, PollEvent};
+use core::ops::ControlFlow;
+use raw_window_handle::HasRawWindowHandle;
+use slint::platform::{Key, WindowAdapter, WindowEvent};
+use slint::{PlatformError, SharedString};
+use std::cell::Cell;
+use std::rc::Rc;
+use std::sync::{Arc, Mutex};
+
+pub struct AndroidPlatform {
+    app: AndroidApp,
+    window: Rc<AndroidWindowAdapter>,
+    event_listener: Option<Box<dyn Fn(&PollEvent<'_>)>>,
+}
+
+impl AndroidPlatform {
+    /// Instantiate a new Android backend given the [`android_activity::AndroidApp`]
+    ///
+    /// Pass the returned value to [`slint::platform::set_platform()`]
+    ///
+    /// # Example
+    /// ```
+    /// #[cfg(target_os = "android")]
+    /// #[no_mangle]
+    /// fn android_main(app: i_slint_backend_android_activity::AndroidApp) {
+    ///     slint::platform::set_platform(Box::new(
+    ///         i_slint_backend_android_activity::AndroidPlatform::new(app),
+    ///     ))
+    ///     .unwrap();
+    ///     // ... your slint application ...
+    /// }
+    /// ```
+    pub fn new(app: AndroidApp) -> Self {
+        Self {
+            app: app.clone(),
+            window: Rc::<AndroidWindowAdapter>::new_cyclic(|w| AndroidWindowAdapter {
+                app,
+                window: slint::Window::new(w.clone()),
+                renderer: i_slint_renderer_skia::SkiaRenderer::default(),
+                event_queue: Default::default(),
+                pending_redraw: Default::default(),
+            }),
+            event_listener: None,
+        }
+    }
+
+    /// Instantiate a new Android backend given the [`android_activity::AndroidApp`]
+    /// and a function to process the events.
+    ///
+    /// This is the same as [`AndroidPlatform::new()`], but it allow you to get notified
+    /// of events.
+    ///
+    /// Pass the returned value to [`slint::platform::set_platform()`]
+    ///
+    /// # Example
+    /// ```
+    /// #[cfg(target_os = "android")]
+    /// #[no_mangle]
+    /// fn android_main(app: i_slint_backend_android_activity::AndroidApp) {
+    ///     slint::platform::set_platform(Box::new(
+    ///         i_slint_backend_android_activity::AndroidPlatform::new_with_event_listener(
+    ///             app,
+    ///             |event| { eprintln!("got event {event:?}") }
+    ///         ),
+    ///     ))
+    ///     .unwrap();
+    ///     // ... your slint application ...
+    /// }
+    /// ```
+    pub fn new_with_event_listener(
```

Comment:
Perhaps it would make sense to make this not a "listener" but an event filter. So a function that we say will be called before we process the event, and the provider can choose to "consume" the event (i.e. prevent it from reaching Slint).

Reply:
I wanted to call filter, but filter doesn't really make more sense. 
The event that are there cannot really be filtering without introducing bugs.
What would be an example of an use case of filtering?

This also doesn't include the input event that are handled by Slint only.

Reply:
The main example I can think of are new events that we don't handle. Such as the save/restore for example.